nengo/base.py

```diff
@@ -54,14 +56,35 @@ class NengoObject(with_metaclass(NetworkMember, SupportDefaultsMixin)):
     seed = IntParam('seed', default=None, optional=True)
 
     def __init__(self, label, seed):
+        super(NengoObject, self).__init__()
         self.label = label
         self.seed = seed
 
     def __getstate__(self):
-        raise NotImplementedError("Nengo objects do not support pickling")
+        state = super(NengoObject, self).__getstate__()
+        del state['_initialized']
+        return state
 
     def __setstate__(self, state):
-        raise NotImplementedError("Nengo objects do not support pickling")
+        from nengo.network import Network
+        super(NengoObject, self).__setstate__(state)
+        setattr(self, '_initialized', True)
+        if len(Network.context) > 0:
+            warnings.warn(
+                "{obj} was not added to the network. When copying objects, "
+                "use the copy method on the object instead of Python's copy "
+                "module. When unpickling objects, they have to be added to "
+                "networks manually.".format(obj=self),
+                NotAddedToNetworkWarning)
```

Comment:
I don't like the fact that unpickled objects don't get added to networks automatically. It would be nice to be able to pickle a whole complex network structure and unpickle it. That doesn't have to be in this PR, just sayin'.